fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -485,23 +498,28 @@ private CompletableFuture<byte[]> readBlock(@Nonnull IndexInput requestingInput,
             return exceptionalFuture;
         }
         final long id = reference.getId();
-        return agilityContext.instrument(LuceneEvents.Events.LUCENE_READ_BLOCK, blockCache.asMap().computeIfAbsent(ComparablePair.of(id, block), ignore -> {
-                    if (sharedCache == null) {
-                        return readData(id, block);
-                    }
-                    final byte[] fromShared = sharedCache.getBlockIfPresent(id, block);
-                    if (fromShared != null) {
-                        agilityContext.increment(LuceneEvents.Counts.LUCENE_SHARED_CACHE_HITS);
-                        return CompletableFuture.completedFuture(fromShared);
-                    } else {
-                        agilityContext.increment(LuceneEvents.Counts.LUCENE_SHARED_CACHE_MISSES);
-                        return readData(id, block).thenApply(data -> {
-                            sharedCache.putBlockIfAbsent(id, block, data);
-                            return data;
-                        });
-                    }
+        try {
+            return agilityContext.instrument(LuceneEvents.Events.LUCENE_READ_BLOCK, blockCache.get(ComparablePair.of(id, block), () -> {
```

Comment:
Please note the change to use `blockCache.get()` instead of `asMap.computeIfAbsent`